distributed/client.py

```diff
@@ -807,6 +820,120 @@ class VersionsDict(TypedDict):
     client: dict[str, dict[str, Any]]
 
 
+_T_LowLevelGraph: TypeAlias = dict[Key, tuple]
+
+
+def is_nested(iterable):
+    for item in iterable:
+        if (
+            isinstance(item, Iterable)
+            and not isinstance(item, str)
+            and not isinstance(item, bytes)
+        ):
+            return True
+    return False
+
+
+class MapLayer(Layer):
+    def __init__(
+        self,
+        func: Callable,
+        iterables: Iterable[Any],
+        key: str | Iterable[str] | None = None,
+        pure: bool = True,
+        annotations: dict[str, Any] | None = None,
+        **kwargs,
+    ):
+        self.func: Callable = func
```

Comment:
nit: right now we're actually annotating types of class attributes at class level, e.g.


```python
class MapLayer:
    func: Callable
    ....
```